src/nwp_consumer/internal/inputs/ecmwf/client.py

```diff
@@ -0,0 +1,243 @@
+    "tas": internal.OCFShortName.TemperatureAGL.value,
+    "uas": internal.OCFShortName.WindUComponentAGL.value,
+    "vas": internal.OCFShortName.WindVComponentAGL.value,
+    "dsrp": internal.OCFShortName.DirectSolarRadiation.value,
+    "uvb": internal.OCFShortName.DownwardUVRadiationAtSurface.value,
+    "hcc": internal.OCFShortName.HighCloudCover.value,
+    "mcc": internal.OCFShortName.MediumCloudCover.value,
+    "lcc": internal.OCFShortName.LowCloudCover.value,
+    "ssrd": internal.OCFShortName.DownwardShortWaveRadiationFlux.value,
+    "strd": internal.OCFShortName.DownwardLongWaveRadiationFlux.value,
+    "tprate": internal.OCFShortName.RainPrecipitationRate.value,
+    "sd": internal.OCFShortName.SnowDepthWaterEquivalent.value,
+    "u100": internal.OCFShortName.WindUComponent100m.value,
+    "v100": internal.OCFShortName.WindVComponent100m.value,
+    "u200": internal.OCFShortName.WindUComponent200m.value,
+    "v200": internal.OCFShortName.WindVComponent200m.value,
+}
+
+# Mapping from ECMWF eccode to ECMWF short name
+# * https://codes.ecmwf.int/grib/param-db/?filter=All
+PARAMETER_ECMWFCODE_MAP: dict[str, str] = {
+    "167.128": "tas",  # 2 metre temperature
+    "165.128": "uas",  # 10 metre U-component of wind
+    "166.128": "vas",  # 10 metre V-component of wind
+    "47.128": "dsrp",  # Direct solar radiation
+    "57.128": "uvb",  # Downward uv radiation at surface
+    "188.128": "hcc",  # High cloud cover
+    "187.128": "mcc",  # Medium cloud cover
+    "186.128": "lcc",  # Low cloud cover
+    "164.128": "clt",  # Cloud area fraction
+    "169.128": "ssrd",  # Surface shortwave radiation downward
+    "175.128": "strd",  # Surface longwave radiation downward
+    "260048": "tprate", # Total precipitation rate
+    "141.128": "sd",    # Snow depth, m
+    "246.228": "u100",  # 100 metre U component of wind
+    "247.228": "v100",  # 100 metre V component of wind
+    "239.228": "u200",  # 200 metre U component of wind
+    "240.228": "v200",  # 200 metre V component of wind
+}
+
+AREA_MAP: dict[str, str] = {
+    "uk": "60/-12/48/3",
+    "eu": "E",
+    "global": "G",
+}
+
+COORDINATE_ALLOW_LIST: typing.Sequence[str] = (
+    "time", "step", "latitude", "longitude"
+)
+
+
+class MARSClient(internal.FetcherInterface):
+    """Implements a client to fetch data from ECMWF's MARS API."""
+
+    server: ecmwfapi.api.ECMWFService
+    area: str
+
+    def __init__(self, area: str) -> None:
+        """Create a new ECMWFMarsClient."""
+        self.server = ECMWFService("mars")
+
+        if area not in AREA_MAP:
+            raise KeyError(f"area must be one of {list(AREA_MAP.keys())}")
+
+        self.area = area
+
+    def listRawFilesForInitTime(self, *, it: dt.datetime) -> list[internal.FileInfoModel]:
+        # For the model we are pulling from, there are only files for 00:00 and 12:00
+        # * Hence, only check requests for these times
+
+        if it.hour not in [0, 12]:
+            return []
```

Comment:
add a logger warning?

Reply:
Done in https://github.com/openclimatefix/nwp-consumer/commit/181bf9789f7cccfc720d3a56d0a807a5d6a4cf26